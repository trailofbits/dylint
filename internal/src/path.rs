pub const ENV_REMOVE_VAR: [&str; 3] = ["std", "env", "remove_var"];
pub const ENV_SET_CURRENT_DIR: [&str; 3] = ["std", "env", "set_current_dir"];
pub const ENV_SET_VAR: [&str; 3] = ["std", "env", "set_var"];
pub const ENV_VAR: [&str; 3] = ["std", "env", "var"];
pub const IO_ERROR: [&str; 4] = ["std", "io", "error", "Error"];
pub const PATH_NEW: [&str; 4] = ["std", "path", "Path", "new"];
pub const PATH_BUF_FROM: [&str; 4] = ["std", "path", "PathBuf", "from"];
