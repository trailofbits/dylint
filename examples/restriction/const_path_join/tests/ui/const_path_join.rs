#![warn(const_path_join)]

use std::path::PathBuf;

fn main() {
    // Test cases with literal strings
    let _ = PathBuf::from("foo").join("bar");
    let _ = PathBuf::from("foo").join("bar").join("baz");
    let _ = PathBuf::new().join("foo").join("bar");

    // Test cases with constant expressions
    let _ = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("target");
    let _ = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("target").join("debug");
    
    // Test cases with mixed literal strings and constant expressions
    let _ = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src").join("lib.rs");
    let _ = PathBuf::from("target").join(env!("CARGO_PKG_NAME")).join("debug");

    // Test cases with camino::Utf8PathBuf
    use camino::Utf8PathBuf;
    let _ = Utf8PathBuf::from("foo").join("bar");
    let _ = Utf8PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("target");
} 