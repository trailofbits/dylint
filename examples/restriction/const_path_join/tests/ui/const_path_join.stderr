warning: path could be constructed from a string literal
  --> const_path_join.rs:8:5
   |
8  |     let _ = PathBuf::from("foo").join("bar");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(const_path_join)]` on by default
   = help: use `PathBuf::from("foo/bar")`

warning: path could be constructed from a string literal
  --> const_path_join.rs:9:5
   |
9  |     let _ = PathBuf::from("foo").join("bar").join("baz");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use `PathBuf::from("foo/bar/baz")`

warning: path could be constructed from a string literal
  --> const_path_join.rs:10:5
   |
10 |     let _ = PathBuf::new().join("foo").join("bar");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use `.join("foo/bar")`

warning: path could be constructed from a string literal
  --> const_path_join.rs:13:5
   |
13 |     let _ = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("target");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use `PathBuf::from(concat!(env!("CARGO_MANIFEST_DIR"), "/target"))`

warning: path could be constructed from a string literal
  --> const_path_join.rs:14:5
   |
14 |     let _ = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("target").join("debug");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use `PathBuf::from(concat!(env!("CARGO_MANIFEST_DIR"), "/target/debug"))`

warning: path could be constructed from a string literal
  --> const_path_join.rs:17:5
   |
17 |     let _ = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src").join("lib.rs");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use `PathBuf::from(concat!(env!("CARGO_MANIFEST_DIR"), "/src/lib.rs"))`

warning: path could be constructed from a string literal
  --> const_path_join.rs:18:5
   |
18 |     let _ = PathBuf::from("target").join(env!("CARGO_PKG_NAME")).join("debug");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use `PathBuf::from(concat!("target", "/", env!("CARGO_PKG_NAME"), "/debug"))`

warning: path could be constructed from a string literal
  --> const_path_join.rs:22:5
   |
22 |     let _ = Utf8PathBuf::from("foo").join("bar");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use `Utf8PathBuf::from("foo/bar")`

warning: path could be constructed from a string literal
  --> const_path_join.rs:23:5
   |
23 |     let _ = Utf8PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("target");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use `Utf8PathBuf::from(concat!(env!("CARGO_MANIFEST_DIR"), "/target"))` 